### ISSUE TITLE
Add activation and cost functions

### ISSUE THREAD

**Issue by KEMBL** [KEMBL/machine-learning#6]
* Softmax (activation) and cross entropy (cost)
* Tanh (activation)

More: https://stackoverflow.com/questions/34229140/choosing-from-different-cost-function-and-activation-function-of-a-neural-networ
